Dockerfile

```diff
@@ -52,20 +90,23 @@ RUN set -ex; \
     yarn install && \
     yarn run build && \
     yarn cache clean && \
-    rm -rf node_modules && \
-    apt-get purge -y --auto-remove yarn curl nodejs gnupg && \
-    rm -rf /var/lib/apt/lists/* /etc/apt/sources.list.d/yarn.list
+    rm -rf node_modules
 
 # Layer cache is pretty much invalidated here all the time,
 # so try not to do anything heavy beyond here.
-FROM $BUILD_BASE AS application
```

Comment:
This was a parameter because we wanted to allow switching the base between `build_admin_ui` and `base`. This was because we wanted to allow skipping the `build_admin_ui` layers when we were building on ARM, because currently the ARM image is built inside an emulator and is really slow. AFAIK github doesn't provide us with ARM vms to speed this up. The trick we used was to build the admin `bundle.js` outside of docker then copy it in for the ARM image. I'd imagine if we are building rust it would take a long time in the emulator so we may want to do something similar.

Reply:
OMG @dbanda thank you for pointing this out!! I haven't gotten into debugging the ARM failure yet but it makes perfect sense.